### PR TITLE
PR merge process

### PR DESCRIPTION
**Pull request by m-nash**
# Description
I wanted to codify the PR process in a place where everyone can reference the same thing as well as we can make agreed upon changes to it over time as needs arise.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by ArcturusZhang — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

File: readme.md [hunk 1]
```diff
@@ -73,15 +73,46 @@ The swagger files are compiled, and then run, which pings the mock server (to ve
...
-These instructions are only a general outline, see [the script](https://github.com/Azure/autorest.csharp/blob/feature/v3/eng/UpdateAzureSdkForNet.ps1) for details:
-
-- Use `dotnet pack` to package up a version of the generator
-- `dotnet pack -o directory` (where directory is replaced with the name of the directory)
-- Copy created nuget package to your local nuget source
-- Update package.props
-- `dotnet restore -S directory` (where directory is replaced with the name of the directory)
+### Validate generator changes against Azure SDK before merging your autorest.csharp PR
+
+When the automatic PR is created for azure-sdk-for-net if there are any issues found all other changes to autorest.csharp are blocked until those issues are resolved.  This means we should be certain that the changes we are making create the expected result in azure-sdk-for-net prior to merging our PR.  In time we should build this into the autorest.csharp CI such that it happens automatically, but in the meantime a few manual steps can help avoid any delays.
```
Comment (lirenhe) [inline readme.md:78]: As @AlexanderSher  is working on the CI improvement,  do we still need those steps after the CI work completes?

**Review by lirenhe — APPROVED** [review]
LGTM.
We would add some additional steps for DPG on how to run tests and generate DPG only projects, and we could do that after this PR get merged.  Thanks.

**Review by jsquire — APPROVED** [review]
I agree fully; this is something that we should be following until CI is able to validate changes across repositories.

File: readme.md (hunk 1 above)
Comment (m-nash) [reply readme.md:78]: I have no idea @AlexanderSher hasn't shared that with the broader community yet, however I do know it is needed now and after any of those changes get deployed we can always adjust this process then

File: readme.md [hunk 2]
```diff
@@ -73,15 +73,46 @@ The swagger files are compiled, and then run, which pings the mock server (to ve
...
+
+- Use `dotnet pack /p:OfficialBuildId=yyyyMMdd.100` to package up a version of the generator
+  - Since there can be real versions with the same date starting at `.100` can ensure no conflicts with official builds.
+  - This will place a nupkg file at this location `[RepoRoot]\artifacts\packages\Debug`
+- Edit the nuget configuration file [here](https://github.com/Azure/azure-sdk-for-net/blob/main/NuGet.Config) to have a new source `<add key="DEBUG" value="[RepoRoot]\artifacts\packages\Debug" />`
+  - This is not intended to be checked in but is a temporary change so `dotnet restore` can find the package
+- Update the [Packages.Data.props](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L156) file to have Microsoft.Azure.AutoRest.Csharp use the new local version you created in step 1
+  - This is not intended to be checked in but is a temporary change so `dotnet restore` can find the package
```
[the remark this answers is not in the document]
Comment (m-nash) [reply readme.md:86]: It not a copy paste error the same comment applies to both steps, they are both intended to be temporary and allow `dotnet restore` to find the local package

**Pull request merged**